text/0000-raw-pointer-comparisons.md

```diff
@@ -0,0 +1,58 @@
+- Feature Name: raw-pointer-comparisons
+- Start Date: 2015-05-27
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+Allow equality, but not order, comparisons between fat raw pointers
+of the same type.
+
+# Motivation
+
+Currently, fat raw pointers can't be compared via either PartialEq or
+PartialOrd (currently this causes an ICE). It seems to me that a primitive
+type like a fat raw pointer should implement equality in some way.
+
+However, there doesn't seem to be a sensible way to order raw fat pointers
+unless we take vtable addresses into account, which is relatively weird.
```

Comment:
`PartialOrd` makes much more sense for raw slices `*const [T]` than for trait pointers, and the ordering is quite clear - (begin1, len1) < (begin2, len2).
(Speaking of weirdness, ordering for unrelated thin pointers doesn't make much sense too (C++, for example, doesn't even define it), but it exists just because having some ordering is useful.)